### PR TITLE
Document the Versioned App Registration

### LINKED ISSUE #1933: Document App Registry's versioning support
As a developer, I'd like to document the versioning support added to the App Registry.

**Acceptance:**
- Add a new section under "[Stream Lifecycle with SKipper](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.M3/reference/htmlsingle/#spring-cloud-dataflow-stream-lifecycle-skipper)"; perhaps call it "Application Registry"
- Expand on the registry support and the versioning relevance.
- Include examples of multiple version registrations for the same application.
- Explain the use of `app default` command.
- Explain the validation support and its applicability for "[update](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.M3/reference/htmlsingle/#spring-cloud-dataflow-stream-lifecycle-skipper-update)" and "[rollback](https://docs.spring.io/spring-cloud-dataflow/docs/1.3.0.M3/reference/htmlsingle/#spring-cloud-dataflow-stream-lifecycle-skipper-rollback)" workflows.

### PR DESCRIPTION
**Pull request by tzolov**
Resolves #1933

 - Add a new section ('Register a Versioned Stream App') under 'Stream Lifecycle with Skipper'.
 - Explain the registry support and the versioning relevance.
 - Provide examples of multiple version registrations for the same application.
 - Explain the use of app default command.
 - Explain the validation support and its applicability for 'update' and 'rollback' workflows.

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 1]
```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
...
+╔══════════════════╤═════════╤════╤════╗
+║     source       │processor│sink│task║
+╠══════════════════╪═════════╪════╪════╣
+║> mysource-0.0.1 <│         │    │    ║
+║mysource-0.0.2    │         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+
+Depends on the schema the provided a URI format should conform to the following:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: Depends -> Depending

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 2]
```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
...
+----
+http://<web-path>/<artifactName>-<version>.jar
+----
+* file schema
+[source,bash]
+----
+file:///<local-path>/<artifactName>-<version>.jar
+----
+
+You can mark the default app version to be used by when deploying Streams. By default the first registered app is marked as default.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: we could drop "by".

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 3]
```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
...
+dataflow:>app unregister --name mysource --type source --version 0.0.1
+dataflow:>app list --id source:mysource
+╔══════════════════╤═════════╤════╤════╗
+║     source       │processor│sink│task║
+╠══════════════════╪═════════╪════╪════╣
+║> mysource-0.0.2 <│         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+When the version is not specified the app registration version marked as default is deleted. Note: You have to manually set another version as default!
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: What will happen if you don't pick another version? We could clarify that maybe.

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 4]
```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
...
+
+Deployed or updated streams use the App versions marked as default unless a different version is provided with the set in the deployment properties.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: >  unless a different version is provided with the set in the deployment properties.

We could drop "with the set"

Comment (sabbyanandan) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: Maybe also add an example of this experience? 

something like:

> stream create foo --definition "mysource | log" (default here is 0.0.2 based on the previous actions ^^)
> stream update foo --properties version.mysource=0.0.3

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 5]
```diff
@@ -869,6 +869,85 @@ The generated package is uploaded to Skipper's package repository and Data Flow
 Skipper to install the package that corresponds to the Stream.
 Subsequent commands to upgrade and rollback applications within the Stream are passed through to Skipper after some validation checks are performed by Data Flow.
 
+=== Register a Versioned Stream App
+Skipper extends the _<<streams.adoc#spring-cloud-dataflow-register-stream-apps, Register a Stream App>>_
+ lifecycle with support of multi-versioned Stream App.
```
[the remark this answers is not in the document]
Comment (tzolov) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: +1

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc (hunk 1 above)
Comment (tzolov) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: +1

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc (hunk 2 above)
Comment (tzolov) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: +1

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc (hunk 3 above)
Comment (tzolov) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: +1

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc (hunk 4 above)
Comment (tzolov) [reply spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: +1

File: spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc [hunk 6]
```diff
@@ -869,6 +869,120 @@ The generated package is uploaded to Skipper's package repository and Data Flow
...
+╠══════════════════╪═════════╪════╪════╣
+║> mysource-0.0.1 <│         │    │    ║
+║mysource-0.0.2    │         │    │    ║
+║mysource-0.0.3    │         │    │    ║
+╚══════════════════╧═════════╧════╧════╝
+----
+
+Depending on the schema the provided a URI should conform to the following formats:
+
+* maven schema
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc]: I think we need to add `docker` schema here

**Comment by ilayaperumalg** [comment]
LGTM

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as e6f7f5a7

**Pull request closed without merging**